icicle/tests/test_field_api.cpp

```diff
@@ -1054,6 +1056,31 @@ TEST_F(FieldApiTestBase, CpuProgramExecutorReturningVal)
   ASSERT_EQ(0, memcmp(out_element_wise.get(), out_vec_ops.get(), total_size * sizeof(scalar_t)));
 }
 
+TEST_F(FieldApiTestBase, Sumcheck)
+{
+  int mle_poly_size = 1 << 13;
+  int nof_mle_poly = 4;
+  scalar_t claimed_sum = scalar_t::from(8);
+
+  // create transcript_config
+  SumcheckTranscriptConfig<scalar_t> transcript_config; // TODO Miki: define labels?
+
+  // create sumcheck
+  auto sumcheck = Sumcheck<scalar_t>::create(claimed_sum, std::move(transcript_config));
+
+  // generate inputs
+  std::vector<std::vector<scalar_t>*> mle_polynomials(nof_mle_poly);
+  for (auto& mle_poly_ptr : mle_polynomials) {
+    mle_poly_ptr = new std::vector<scalar_t>(mle_poly_size);
```

Comment:
avoid new at all cost. This is a bad thing and indeed your test is leaking memory since you don't call delete.
ALways prefer smart pointer and even better if you can make the vector pointer jsut a vector, and move it into the outer vector. It's zero cost almost